capplets/appearance/appearance-style.c

```diff
@@ -152,6 +153,34 @@ treeview_selection_changed_callback (GtkTreeSelection *selection, guint data)
 
     if (list_value) {
       g_settings_set_string (settings, key, list_value);
+
+    /*Load the gnome interface schema if we are running under wayland and it is present*/
+      if (!(GDK_IS_X11_DISPLAY (gdk_display_get_default())))
+      {
+        GSettingsSchemaSource *source = g_settings_schema_source_get_default ();
+
+        if (source)
+        {
+          GSettingsSchema *schema = g_settings_schema_source_lookup (source, INTERFACE_GNOME_SCHEMA, TRUE);
+          {
+            if (schema)
+            {
+              GSettings *interface_gnome_settings = NULL;
+              interface_gnome_settings = g_settings_new_full (schema, NULL, NULL);
+
+              g_message("%s", key);
+
+              if (strcmp (key, ICON_THEME_KEY) == 0)
+                g_settings_set_string (interface_gnome_settings,
+                               ICON_THEME_KEY, list_value);
+
+              if (strcmp (key, GTK_THEME_KEY) == 0)
+                g_settings_set_string (interface_gnome_settings,
+                               GTK_THEME_KEY, list_value);
```

Comment:
```suggestion
              if (strcmp (key, ICON_THEME_KEY) == 0 ||
                  strcmp (key, GTK_THEME_KEY) == 0)
                g_settings_set_string (interface_gnome_settings, key, list_value);
```

Reply:
Nice improvement here, should allow all of the usable theme customizations (gsettings does not control window border theme in wayfire etc) to be applied using less code

Reply:
https://github.com/mate-desktop/mate-control-center/pull/757/commits/b29a2e863965cea9f161221835b652758c95f80b switches the logic here to exclude the window border theme keys (which are not used in wayfire or similar compositors) and apply the others